### DISCUSSION TITLE
Rocketmq client 客服连接报错（C#）

### DISCUSSION THREAD

**Discussion by sandro-123** [apache/rocketmq#8867] [General]
服务器：
     操作系统：win10
     JAVA版本：23.0.1
     Rocketmq: 5.3.1
     IP :172.18.47.40
服务器配置后，使用MQADMIN可以正常收发消息，创建队列等。但是使用官方的C#代码测试时
报错RpcException: Status(StatusCode="Internal", Detail="Error starting gRPC call错误。查看
服务器日志发现客户端已经和服务服务器建立连接，只是建立连接握手失败。
启动命令
start bin\mqnamesrv.cmd
start bin\mqbroker.cmd -n 172.18.47.40:9876 -c conf\broker.conf

**brock 配置**
![配置](https://github.com/user-attachments/assets/d19d4c3d-b2ca-4029-9125-1ac346d5a7b9)
**ACL 配置**
![acl](https://github.com/user-attachments/assets/66e6d316-daca-44c3-a94a-82fe754b86f3)
**服务日志：**
![日志](https://github.com/user-attachments/assets/b0539669-7226-4c2a-810e-7dc670b8201f)
**代码**
![代码](https://github.com/user-attachments/assets/6deeefa2-3467-4fce-b859-3e33d1e8db52)
**错误代码**：

System.AggregateException
  HResult=0x80131500
  Message=One or more errors occurred. (Status(StatusCode="Internal", Detail="Error starting gRPC call. HttpRequestException: Requesting HTTP version 2.0 with version policy RequestVersionOrHigher while unable to establish HTTP/2 connection.", DebugException="System.Net.Http.HttpRequestException: Requesting HTTP version 2.0 with version policy RequestVersionOrHigher while unable to establish HTTP/2 connection."))
  Source=System.Private.CoreLib
  StackTrace:
   在 System.Threading.Tasks.Task.ThrowIfExceptional(Boolean includeTaskCanceledExceptions)
   在 System.Threading.Tasks.Task.Wait(Int32 millisecondsTimeout, CancellationToken cancellationToken)
   在 System.Threading.Tasks.Task.Wait()
   在 examples.QuickStart.Main() 在 E:\temp\rocketmq-client-csharp-5.2.0-rc1-source-release\csharp\examples\QuickStart.cs 中: 第 33 行

  此异常最初是在此调用堆栈中引发的: 
    [外部代码]

内部异常 1:
RpcException: Status(StatusCode="Internal", Detail="Error starting gRPC call. HttpRequestException: Requesting HTTP version 2.0 with version policy RequestVersionOrHigher while unable to establish HTTP/2 connection.", DebugException="System.Net.Http.HttpRequestException: Requesting HTTP version 2.0 with version policy RequestVersionOrHigher while unable to establish HTTP/2 connection.")

内部异常 2:
HttpRequestException: Requesting HTTP version 2.0 with version policy RequestVersionOrHigher while unable to establish HTTP/2 connection.

**Comment by sandro-123** [comment]
已经解决了。

**Reply by carson-repository** [reply]
如何解决的
